transcriptic/jupyter/container.py

```diff
@@ -0,0 +1,172 @@
+    well_map: dict
+        Well mapping with well indices for keys and well names as values
+    aliquots: DataFrame
+        DataFrame of aliquots present in the container. DataFrame index
+        now corresponds to the Well Index.
+    container_type: autoprotocol.container_type.ContainerType
+        Autoprotocol ContainerType object with many useful container type
+        information and functions.
+    cover: str
+        Cover type of container
+    storage: str
+        Storage condition of container
+
+        Example Usage:
+
+        .. code-block:: python
+
+          my_container = container("ct186apgz6a374")
+
+          my_container.well_map
+
+          my_container.container_type.col_count
+          my_container.container_type.robotize("B1")
+          my_container.container_type.humanize(12)
+
+
+    """
+
+    def __init__(self, container_id, attributes=None, connection=None):
+        """
+        Initialize a Container by providing a container name/id. The attributes and
+        connection parameters are generally not specified unless one wants to manually
+        initialize the object.
+
+        Parameters
+        ----------
+        container_id: str
+            Container name or id in string form
+        attributes: Optional[dict]
+            Attributes of the container
+        connection: Optional[transcriptic.config.Connection]
+            Connection context. The default context object will be used unless
+            explicitly provided
+        """
+        super(Container, self).__init__(
+            "container", container_id, attributes, connection
+        )
+        # TODO: Unify container "label" with name, add Containers route
+        self.id = container_id
+        self.cover = self.attributes["cover"]
+        self.name = self.attributes["label"]
+        self.storage = self.attributes["storage_condition"]
+        self.well_map = {
+            aliquot["well_idx"]: aliquot["name"]
+            for aliquot in self.attributes["aliquots"]
+        }
+        self.container_type = self._parse_container_type()
+        self._aliquots = pd.DataFrame()
+
+    def _parse_container_type(self):
+        """Helper function for parsing container string into container object"""
+
+        container_type = self.attributes["container_type"]
+
+        # Return the corresponding AP-Py container object for now. In the future,
+        # consider merging the current and future dictionary when instantiating
+        # container_type
+        try:
+            from autoprotocol.container_type import _CONTAINER_TYPES
+
+            return _CONTAINER_TYPES[container_type["shortname"]]
+        except ImportError:
+            warnings.warn(
+                "Please install `autoprotocol-python` in order to get container types"
+            )
+            return None
+        except KeyError:
+            warnings.warn("ContainerType given is not supported yet in AP-Py")
+            return None
```

Comment:
Will there be a case where container_type is Null? Should it be caught in exception?

Reply:
yup, we may want to catch it.. I'm inclined to leave this as a pure refactor for now though..

Reply:
kk - sounds good. thanks!